src/protocols/secure_channel/CASESession.cpp

```diff
@@ -151,19 +151,26 @@ class CASESession::WorkHelper
     // The `status` value is the result of the work callback (called beforehand).
     typedef CHIP_ERROR (CASESession::*AfterWorkCallback)(DATA & data, CHIP_ERROR status);
 
+    // This callback will be called if WorkCallback returns error code other than CHIP_NO_ERROR,
```

Comment:
I think it's fine to have a callback for if/when the scheduling fails, but it shouldn't occur if the work is scheduled but itself returns a (work-specific) error code that isn't OK. Those errors from work are normally (when scheduling works) ferried back to the CASESession object via the 'C' part of the work, and handled there. Arguably this callback should just be for if scheduling back to the fg thread fails.

Reply:
Sure, thanks for the clarification, will change accordingly.